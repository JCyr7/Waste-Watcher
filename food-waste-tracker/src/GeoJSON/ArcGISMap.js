import React, {Component} from 'react'
import {Dimensions, StyleSheet, View} from 'react-native'
import MapView, {Geojson, PROVIDER_GOOGLE} from 'react-native-maps'
import mapStyle from './mapStyle.json'
import geojson from './example.json'

//Component displays a map view of Maine
export default class ArcGISMap extends Component {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      // Map container view
      <View style={styles.mapContainer}>
        <MapView
          style={styles.map}
          width={Dimensions.get('window').width}
          height={Dimensions.get('window').height}
          zoomEnabled={true}
          provider={PROVIDER_GOOGLE}
          customMapStyle={mapStyle}
          initialRegion={{
            latitude: 39.2538,
            longitude: -68.4455,
            latitudeDelta: 5,
            longitudeDelta: 5
          }}>
          {/* GeoJSON object displays simple line for testing */}
          <Geojson geojson={geojson} />
        </MapView>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  mapContainer: {
    borderRadius: 10,
    overflow: 'hidden',
    width: '100%', // Added width to ensure it takes up the full width of its container
    aspectRatio: 1.5, // Adjust the aspect ratio as needed
  },
  map: {
    overflow: 'hidden',
    height: '100%',
    width: '100%'
  }
})
