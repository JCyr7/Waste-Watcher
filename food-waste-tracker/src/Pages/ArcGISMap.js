import React, {Component} from 'react'
import {Dimensions, StyleSheet, View} from 'react-native'
import MapView, {Geojson, PROVIDER_GOOGLE} from 'react-native-maps'
import mapStyle from '../GeoJSON/mapStyle.json'
import geojson from '../GeoJSON/example.json'

//Component displays a map view of Maine
export default class ArcGISMap extends Component {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      // Map container view
      <View style={styles.mapContainer}>
        <MapView
          style={styles.map}
          width={Dimensions.get('window').width}
          height={Dimensions.get('window').height}
          zoomEnabled={true}
          provider={PROVIDER_GOOGLE}
          customMapStyle={mapStyle}
          initialRegion={{
            latitude: 45.2538,
            longitude: -69.2455,
            latitudeDelta: 5,
            longitudeDelta: 5
          }}>
          {/* GeoJSON object displays simple line for testing */}
          <Geojson geojson={geojson} />
        </MapView>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  mapContainer: {
    borderRadius: 10,
    overflow: 'hidden'
  },
  map: {
    overflow: 'hidden',
    height: '100%',
    width: '100%'
  }
})
