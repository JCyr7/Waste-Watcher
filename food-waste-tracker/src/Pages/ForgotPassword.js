import * as React from 'react'
import {Component} from 'react'
import { LinearGradient } from 'expo-linear-gradient';
import {
  KeyboardAvoidingView,
  StyleSheet,
  Image,
  TextInput,
  Text,
  Pressable,
  TouchableOpacity
} from 'react-native'
import {COLORS} from '../Utils/colors'

export default class ForgotPassword extends Component {
  constructor(props) {
    super(props)
    this.state = {
      invalidEmail: true,
      email: ''
    }
  }

  // Validate text method ensure that email field is not empty --> try and immplement valid email checking --> query database for valid email
  validateText() {
    if (this.state.email === '') {
      this.setState({invalidEmail: true})
    } else {
      this.setState({invalidEmail: false})
    }
  }

  // Sets the value of the email state, and calls the validate text method
  setText(value) {
    // console.log(this.state.email);
    this.validateText()
    this.setState({email: value})
  }

  render() {
    const { navigation } = this.props;

    return (
      <LinearGradient
        colors={[COLORS.blue, COLORS.green]}
        style={styles.container}
        start={{ x: 0, y: 0.2 }}
        end={{ x: 1, y: 1 }}>

        {/* Container for page */}
        <KeyboardAvoidingView behavior='padding' style={styles.container}>
          <Pressable
            onPress={() => navigation.navigate('LogoutPage')}
            style={({ pressed }) => [
              {
                backgroundColor: pressed
                  ? COLORS.transparent
                  : COLORS.transparent,
              },
              styles.backButton,
            ]}>
            <Text style={styles.backButtontext}>Ã—</Text>
          </Pressable>

          <Image
            style={styles.image}
            source={require('../../images/FoodRescueMaine_Logo_Final-01.png')}
          />
          <Text style={styles.header}>Forgot Password?</Text>
          <Text style={styles.subheader}>Please enter your email and we'll send you a link to reset your password.</Text>


          <TextInput
            cursorColor={'white'}
            selectionColor={'white'}
            placeholder='example@gmail.com'
            placeholderTextColor={COLORS.white} // Set the color of the placeholder text
            style={[styles.input, { color: COLORS.white }]}             
            onChangeText={(value) => this.setText(value)}
            onEndEditing={() => this.validateText()}>
          </TextInput>

          <Pressable
            //onPress={() => navigation.navigate('LogoutPage')}
            style={({ pressed }) => [
              {
                backgroundColor: pressed ? COLORS.whitetransparent : COLORS.transparent
              },
              styles.submitButton,
            ]}>
            <Text style={styles.submitButtonTextColor}>Reset Password</Text>
          </Pressable>
        </KeyboardAvoidingView>
      </LinearGradient>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    padding: 0,
    margin: 0,
    width: '100%',
    height: '100%',
  },
  image: {
    flex: .1, 
    width: '100%', // Use 100% width to ensure it doesn't exceed the screen width
    marginBottom: '10%',
    aspectRatio: 792 / 283, // Set the aspect ratio based on your image dimensions
  },
  backButton: {
    alignSelf: 'flex-start',
    margin: 10,
    marginTop: 40,
    borderRadius: 14,
  },
  backButtontext: {
    fontSize: 40,
    color: COLORS.white,
    paddingTop: 20,
    paddingBottom: 20,
    paddingLeft: 25,
    paddingRight: 25
  },
  header: {
    fontSize: 30,
    fontWeight: 'bold',
    color: COLORS.white
  },
  subheader: {
    color: COLORS.white,
    marginLeft: '.5%',
    marginRight: '7.5%',
    fontSize: 12
  },
  input: {
    height: 50,
    width: '85%',
    borderRadius: 7,
    backgroundColor: COLORS.whitetransparent,
    margin: 8,
    paddingLeft: 15,
    fontSize: 15,
  },
  submitButton: {
    height: 50,
    width: '85%',
    alignItems: 'center',
    justifyContent: 'space-evenly',
    borderColor: COLORS.whitetransparent,
    borderWidth: 3,
    borderRadius: 7,
    margin: 8
  },
  submitButtonTextColor: {
    color: 'white',
    fontSize: 15,
  }
})
