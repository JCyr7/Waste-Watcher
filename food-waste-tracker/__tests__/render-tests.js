/* eslint-disable no-undef */
import React from 'react'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react';
import ArcGISMap from '../src/Pages/ArcGISMap'
import ForgotPassword from '../src/Pages/ForgotPassword'
import HomePage from '../src/Pages/HomePage'
import LeaderBoardPage from '../src/Pages/LeaderboardPage'
//import Login from '../src/Pages/LoginPage'
import LogoutPage from '../src/Pages/LogoutPage'
import ProfilePage from '../src/Pages/ProfilePage'

jest.mock('@react-native-async-storage/async-storage')

test('renders correctly', () => {
  const tree = renderer.create(<LogoutPage />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders correctly', () => {
  const tree = renderer.create(<ForgotPassword />).toJSON()
  expect(tree).toMatchSnapshot()
})

const mockObject = {
  getItem: jest.fn()
}

/*
describe('HomePage', () => {
  it('calls getItem on the object', () => {
    // Render the component
    renderer.create(<HomePage object={mockObject} />);

    // Check that getItem was called
    expect(mockObject.getItem).toMatchSnapshot();
  }); // Add closing parenthesis here
});/*


/*test('renders correctly', () => {
  const tree = renderer.create(<LeaderBoardPage />).toJSON()
  expect(tree).toMatchSnapshot()
})*/

/*test('renders correctly', () => {
  const tree = renderer.create(<Login />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders correctly', () => {
  const tree = renderer.create(<ArcGISMap />).toJSON()
  expect(tree).toMatchSnapshot()
})*/

test('renders correctly', () => {
  const tree = renderer.create(<ProfilePage />).toJSON()
  expect(tree).toMatchSnapshot()
})

