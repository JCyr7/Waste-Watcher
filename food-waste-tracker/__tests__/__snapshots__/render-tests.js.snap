// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  colors={
    [
      4278216447,
      4278249075,
    ]
  }
  endPoint={
    [
      1,
      1,
    ]
  }
  startPoint={
    [
      0,
      0.2,
    ]
  }
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "height": "100%",
      "margin": 0,
      "padding": 0,
      "width": "100%",
    }
  }
>
  <RCTSafeAreaView
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "height": "100%",
        "margin": 0,
        "padding": 0,
        "width": "100%",
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "aspectRatio": 2.7985865724381624,
          "flex": 0.25,
          "marginBottom": "10%",
          "marginTop": 20,
          "width": "100%",
        }
      }
    />
    <Text
      style={
        {
          "color": "white",
          "fontSize": 32,
          "fontWeight": "bold",
        }
      }
    >
      Waste Watcher
    </Text>
    <Text
      style={
        {
          "color": "white",
          "fontSize": 24,
          "marginBottom": "10%",
        }
      }
    >
      Track n' Reduce Food Waste
    </Text>
    <TextInput
      cursorColor="white"
      keyboardType="email-address"
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#fff"
      selectionColor="white"
      style={
        [
          {
            "backgroundColor": "rgba(255, 255, 255, 0.25)",
            "borderRadius": 7,
            "fontSize": 15,
            "height": 50,
            "margin": 8,
            "paddingLeft": 15,
            "width": "85%",
          },
          {
            "color": "#fff",
          },
        ]
      }
    />
    <TextInput
      cursorColor="white"
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#fff"
      secureTextEntry={true}
      selectionColor="white"
      style={
        [
          {
            "backgroundColor": "rgba(255, 255, 255, 0.25)",
            "borderRadius": 7,
            "fontSize": 15,
            "height": 50,
            "margin": 8,
            "paddingLeft": 15,
            "width": "85%",
          },
          {
            "color": "#fff",
          },
        ]
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "backgroundColor": "rgba(0,0,0,0)",
          },
          {
            "alignItems": "center",
            "borderColor": "rgba(255, 255, 255, 0.25)",
            "borderRadius": 7,
            "borderWidth": 3,
            "height": 50,
            "justifyContent": "space-evenly",
            "margin": 8,
            "width": "85%",
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 15,
            "fontWeight": "bold",
          }
        }
      >
        Log In
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "backgroundColor": "rgba(0,0,0,0)",
          },
          {
            "alignItems": "center",
            "borderColor": "rgba(255, 255, 255, 0.25)",
            "borderRadius": 7,
            "borderWidth": 3,
            "height": 50,
            "justifyContent": "space-evenly",
            "margin": 8,
            "width": "85%",
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 15,
            "fontWeight": "bold",
          }
        }
      >
        Bypass Log In
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "backgroundColor": "rgba(0,0,0,0)",
          },
          {
            "alignItems": "center",
            "borderRadius": 7,
            "height": 30,
            "justifyContent": "center",
            "marginTop": 3,
            "paddingTop": 0,
            "width": "85%",
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "white",
          }
        }
      >
        Forgot your login details? 
        <Text
          style={
            {
              "fontWeight": "bold",
            }
          }
        >
          Get help signing in.
        </Text>
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 16,
          "marginBottom": 15,
          "marginTop": 15,
        }
      }
    >
      OR
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 0.25)",
          "borderRadius": 7,
          "flexDirection": "row",
          "fontSize": 15,
          "height": 50,
          "justifyContent": "center",
          "width": "85%",
        }
      }
    >
      <Text />
      <Text
        style={
          {
            "color": "white",
            "fontWeight": "bold",
            "paddingLeft": 8,
          }
        }
      >
        Sign in with Google
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "backgroundColor": "rgba(0,0,0,0)",
          },
          {
            "alignItems": "center",
            "borderRadius": 7,
            "height": 50,
            "justifyContent": "center",
            "marginTop": 10,
            "paddingTop": 0,
            "width": "85%",
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "white",
          }
        }
      >
        Dont have an account? 
        <Text
          style={
            {
              "fontWeight": "bold",
            }
          }
        >
          Sign up
        </Text>
      </Text>
    </View>
  </RCTSafeAreaView>
</ViewManagerAdapter_ExpoLinearGradient>
`;

exports[`renders correctly 2`] = `
<ViewManagerAdapter_ExpoLinearGradient
  colors={
    [
      4278216447,
      4278249075,
    ]
  }
  endPoint={
    [
      1,
      1,
    ]
  }
  startPoint={
    [
      0,
      0.2,
    ]
  }
  style={
    {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "flex": 1,
          "flexDirection": "column",
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flex": 1,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "backgroundColor": "rgba(0,0,0,0)",
            },
            {
              "alignSelf": "flex-start",
              "borderRadius": 14,
              "margin": 10,
              "marginTop": 40,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 40,
              "paddingBottom": 20,
              "paddingLeft": 25,
              "paddingRight": 25,
              "paddingTop": 20,
            }
          }
        >
          ×
        </Text>
      </View>
      <Text
        style={
          {
            "alignItems": "center",
            "color": "white",
            "fontSize": 35,
            "justifyContent": "center",
          }
        }
      >
        Forgot Password?
      </Text>
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 15,
            "marginBottom": 15,
            "marginTop": 80,
            "width": "80%",
          }
        }
      >
        Please enter your email and we'll send you an email with a link to reset your password.
      </Text>
      <TextInput
        cursorColor="white"
        onChangeText={[Function]}
        onEndEditing={[Function]}
        placeholder="example@gmail.com"
        placeholderTextColor="#fff"
        selectionColor="white"
        style={
          [
            {
              "backgroundColor": "rgba(255, 255, 255, 0.25)",
              "borderRadius": 7,
              "fontSize": 15,
              "height": 50,
              "marginBottom": 8,
              "paddingLeft": 15,
              "width": "85%",
            },
            {
              "color": "#fff",
            },
          ]
        }
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "backgroundColor": "rgba(0,0,0,0)",
            },
            {
              "alignItems": "center",
              "borderColor": "rgba(255, 255, 255, 0.25)",
              "borderRadius": 7,
              "borderWidth": 3,
              "height": 50,
              "justifyContent": "space-evenly",
              "margin": 8,
              "width": "85%",
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 15,
            }
          }
        >
          Reset Password
        </Text>
      </View>
    </View>
  </View>
</ViewManagerAdapter_ExpoLinearGradient>
`;

exports[`renders correctly 3`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 0,
        "elevation": 10,
        "height": "20%",
        "justifyContent": "center",
        "marginBottom": 20,
        "marginTop": "0%",
        "shadowColor": "#0066ff",
        "shadowOffset": {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 8,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": "2.5%",
        }
      }
    >
      <Image
        source={1}
        style={
          {
            "height": 50,
            "width": 50,
          }
        }
      />
    </View>
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginVertical": "1%",
        }
      }
    >
      Frank Pug
    </Text>
    <Text
      style={
        {
          "color": "#474747",
          "fontSize": 16,
          "marginBottom": "5%",
        }
      }
    >
      frankdapug
    </Text>
  </View>
  <View
    style={
      {
        "flex": 1,
        "height": "100%",
        "marginTop": "5%",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "backgroundColor": "rgba(0,0,0,0)",
          },
          {
            "alignItems": "center",
            "borderBottomColor": "rgba(255, 255, 255, 0.75)",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": "13%",
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          {
            "marginLeft": "7.5%",
            "marginRight": "2%",
          }
        }
      >
        <Image
          source={1}
          style={
            {
              "height": 35,
              "width": 35,
            }
          }
        />
      </View>
      <View
        style={
          {
            "flex": 1,
            "justifyContent": "center",
            "marginLeft": "3%",
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 19,
              "fontWeight": "bold",
              "marginBottom": 1,
            }
          }
        >
          Notifications
        </Text>
        <Text
          style={
            {
              "color": "#474747",
              "fontSize": 13,
              "marginRight": "10%",
            }
          }
        >
          No new messages
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "backgroundColor": "rgba(0,0,0,0)",
          },
          {
            "alignItems": "center",
            "borderBottomColor": "rgba(255, 255, 255, 0.75)",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": "13%",
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          {
            "marginLeft": "7.5%",
            "marginRight": "2%",
          }
        }
      >
        <Image
          source={1}
          style={
            {
              "height": 35,
              "width": 35,
            }
          }
        />
      </View>
      <View
        style={
          {
            "flex": 1,
            "justifyContent": "center",
            "marginLeft": "3%",
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 19,
              "fontWeight": "bold",
              "marginBottom": 1,
            }
          }
        >
          Friends
        </Text>
        <Text
          style={
            {
              "color": "#474747",
              "fontSize": 13,
              "marginRight": "10%",
            }
          }
        >
          View and invite friends
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "backgroundColor": "rgba(0,0,0,0)",
          },
          {
            "alignItems": "center",
            "borderBottomColor": "rgba(255, 255, 255, 0.75)",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": "13%",
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          {
            "marginLeft": "7.5%",
            "marginRight": "2%",
          }
        }
      >
        <Image
          source={1}
          style={
            {
              "height": 35,
              "width": 35,
            }
          }
        />
      </View>
      <View
        style={
          {
            "flex": 1,
            "justifyContent": "center",
            "marginLeft": "3%",
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 19,
              "fontWeight": "bold",
              "marginBottom": 1,
            }
          }
        >
          Streak
        </Text>
        <Text
          style={
            {
              "color": "#474747",
              "fontSize": 13,
              "marginRight": "10%",
            }
          }
        >
          Track daily, earn rewards
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "backgroundColor": "rgba(0,0,0,0)",
          },
          {
            "alignItems": "center",
            "borderBottomColor": "rgba(255, 255, 255, 0.75)",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": "13%",
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          {
            "marginLeft": "7.5%",
            "marginRight": "2%",
          }
        }
      >
        <Image
          source={1}
          style={
            {
              "height": 35,
              "width": 35,
            }
          }
        />
      </View>
      <View
        style={
          {
            "flex": 1,
            "justifyContent": "center",
            "marginLeft": "3%",
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 19,
              "fontWeight": "bold",
              "marginBottom": 1,
            }
          }
        >
          My Badges
        </Text>
        <Text
          style={
            {
              "color": "#474747",
              "fontSize": 13,
              "marginRight": "10%",
            }
          }
        >
          Earn badges, complete goals
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "backgroundColor": "rgba(0,0,0,0)",
          },
          {
            "alignItems": "center",
            "borderBottomColor": "rgba(255, 255, 255, 0.75)",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": "13%",
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          {
            "marginLeft": "7.5%",
            "marginRight": "2%",
          }
        }
      >
        <Image
          source={1}
          style={
            {
              "height": 35,
              "width": 35,
            }
          }
        />
      </View>
      <View
        style={
          {
            "flex": 1,
            "justifyContent": "center",
            "marginLeft": "3%",
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 19,
              "fontWeight": "bold",
              "marginBottom": 1,
            }
          }
        >
          Settings
        </Text>
        <Text
          style={
            {
              "color": "#474747",
              "fontSize": 13,
              "marginRight": "10%",
            }
          }
        >
          Edit your account info
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "backgroundColor": "rgba(0,0,0,0)",
          },
          {
            "alignItems": "center",
            "borderBottomColor": "rgba(255, 255, 255, 0.75)",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": "13%",
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          {
            "marginLeft": "7.5%",
            "marginRight": "2%",
          }
        }
      >
        <Image
          source={1}
          style={
            {
              "height": 35,
              "width": 35,
            }
          }
        />
      </View>
      <View
        style={
          {
            "flex": 1,
            "justifyContent": "center",
            "marginLeft": "3%",
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 19,
              "fontWeight": "bold",
              "marginBottom": 1,
            }
          }
        >
          Refer a Friend
        </Text>
        <Text
          style={
            {
              "color": "#474747",
              "fontSize": 13,
              "marginRight": "10%",
            }
          }
        >
          Connect and track together
        </Text>
      </View>
    </View>
  </View>
</View>
`;
